test.py

```diff
@@ -346,181 +427,28 @@ async def reap(done, pending, signaled):
         await TestSuite.artifacts.cleanup_before_exit()
 
     console.print_end_blurb()
+    return total_tests, failed_tests
 
 
-def print_summary(failed_tests: List["Test"], cancelled_tests: int, options: argparse.Namespace) -> None:
+def print_summary(failed_tests: List["Test"], cancelled_tests: int, options: argparse.Namespace, failed_pytest_tests: list[SimpleNamespace], total_tests_pytest: int) -> None:
     rusage = resource.getrusage(resource.RUSAGE_CHILDREN)
     cpu_used = rusage.ru_stime + rusage.ru_utime
     cpu_available = (time.monotonic() - launch_time) * multiprocessing.cpu_count()
     utilization = cpu_used / cpu_available
     print(f"CPU utilization: {utilization*100:.1f}%")
-    if failed_tests:
+    if len(failed_tests) > 0 or len(failed_pytest_tests) > 0:
+        all_fails = [*failed_tests, *failed_pytest_tests]
+        total_tests = TestSuite.test_count() + total_tests_pytest
         print("The following test(s) have failed: {}".format(
-            palette.path(" ".join([t.name for t in failed_tests]))))
+            palette.path(" ".join([t.name for t in all_fails]))))
         if options.verbose:
             for test in failed_tests:
                 test.print_summary()
                 print("-"*78)
         if cancelled_tests > 0:
-            print(f"Summary: {len(failed_tests)} of the total {TestSuite.test_count()} tests failed, {cancelled_tests} cancelled")
-        else:
-            print(f"Summary: {len(failed_tests)} of the total {TestSuite.test_count()} tests failed")
-
-
-def summarize_boost_tests(tests):
```

Comment:
It's going to be simpler to review such large change, if large sets of methods/functions/classes/whatever that become unused along the way are removed with a separate trailing patch

Reply:
I can extract deletion to the following PR, but for me, it's logical to remove code in PR where it's not needed anymore.

Reply:
This concern still applies. Can this removal block be moved to the last patch titled "test.py: clean code that isn't used anymore"?

Reply:
done